src/Shared/PropertyFetcher.AOT.cs

```diff
@@ -0,0 +1,236 @@
+// <copyright file="PropertyFetcher.AOT.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#nullable disable
+
+// NOTE: This version of PropertyFetcher is AOT-compatible.
+// Usages of the non-AOT-compatible version can be moved over to this one when they need to support AOT/trimming.
+
```

Comment:
```suggestion
// Copied from https://github.com/open-telemetry/opentelemetry-dotnet/blob/86a6ba0b7f7ed1f5e84e5a6610e640989cd3ae9f/src/Shared/DiagnosticSourceInstrumentation/PropertyFetcher.cs#L30

```

Reply:
I've opened https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1416 to remove the old version and update the other references to this version.